horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/RunServiceTest.java

```diff
@@ -59,31 +68,219 @@
    EventBus eventBus;
 
    @org.junit.jupiter.api.Test
-   public void testDataSetCreated(TestInfo info) throws InterruptedException {
+   public void testStraightforwardTransformationEmptyResult(TestInfo info) throws InterruptedException {
+      BlockingQueue<DataSet> dataSetQueue = eventConsumerQueue(DataSet.class, DataSet.EVENT_NEW);
+      NamedJsonPath path = createExampleNamedJsonPath("foo", "$.value");
+      Schema schema = createExampleSchema(info, true);
+
+      assertNotNull(schema.id);
+      Transformer transformer = createExampleTransformerWithJsonPath("acme", schema, "", path);
+      Test exampleTest = createExampleTest(getTestName(info));
+      Test test = createTest(exampleTest);
+      addTransformer(test, transformer);
+      uploadRun("{\"corporation\":\"acme\"}", test.name);
+
+      DataSet event = dataSetQueue.poll(10, TimeUnit.SECONDS);
+      assertNotNull(event);
+      JsonNode node = event.data;
+      assertTrue(node.isArray());
+      assertTrue(node.isEmpty());
+   }
+
+   private NamedJsonPath createExampleNamedJsonPath(String name, String path) {
+      NamedJsonPath jsonpath = new NamedJsonPath();
+      jsonpath.name = name;
+      jsonpath.jsonpath = path;
+      return jsonpath;
+   }
+
+   @org.junit.jupiter.api.Test
+   public void testTransformationWithoutSchema(TestInfo info) throws InterruptedException {
       BlockingQueue<DataSet> dataSetQueue = eventConsumerQueue(DataSet.class, DataSet.EVENT_NEW);
       Test test = createTest(createExampleTest(getTestName(info)));
-      Schema schema = createExampleSchema(info);
       setTestVariables(test, "Value", "value");
 
-      uploadRun(runWithValue(schema, 42).toString(), test.name);
+      dataSetQueue.drainTo(new ArrayList<>());
```

Comment:
Shouldn't we rather assert that the queue is empty?

Reply:
I don't want other tests to spuriously be the cause of this assertion to fail. Which is what I've seen sometimes.

Reply:
Sure, spurious failures are undesired. An assertion here would make sure we know that it is not *this* test that has a failure; events from other tests should be handled in `@After` methods there.
OK, let's keep it as-is for this and we'll properly deal with any instability later.

Reply:
I have added before and after annotations.